algorithms.py

```diff
@@ -942,36 +943,48 @@ def get_total_gc_rate(self, label):
 
     def get_total_gc_left_rate(self, label):
         gc_left_total = self.get_total_gc_left(label)
-        # NOTE: this is an approximation. Should we be multiplying by the
-        # local GC rate when iterating over the segments?
-        mean_gc_rate = self.gc_map.mean_rate
-        return mean_gc_rate * self.track_length * gc_left_total
-
-    def get_individual_length(self, head):
-        tail = head
-        while tail.next is not None:
-            tail = tail.next
-        # TODO adjust for discrete/continuous genome here?
-        return tail.right - head.left - 1
+        return gc_left_total
 
     def get_total_gc_left(self, label):
         gc_left_total = 0
-        x = (self.track_length - 1) / self.track_length
-        for pop in self.P:
-            for ind in pop.iter_label(label):
-                dist = self.get_individual_length(ind)
-                gc_left_total += 1 - x ** dist
+        if self.gc_rate_constant:
```

Comment:
Let's just get rid of this branch (and the gc_rate_constant flag) for now - you can keep a branch with what you have, but let's get some working code in as soon as possible.